src/pages/test.tsx

```diff
@@ -0,0 +1,89 @@
+// @ts-nocheck
+import React from "react";
+
+import AuthWall from "../auth/AuthWall";
+import {
+  addFacilityCapacity,
+  getMigrationFacilities,
+  getMigrationModelVersions,
+  getMigrationScenarios,
+} from "../database";
+import { totalIncarceratedPopulation } from "../impact-dashboard/EpidemicModelContext";
+import PageInfo from "../site-metadata/PageInfo";
+
+function calculateFacilityCapacity(modelInputs) {
+  const occupancyPct = modelInputs.facilityOccupancyPct;
+  const pop = totalIncarceratedPopulation(modelInputs);
+  if (pop && occupancyPct) {
+    return pop * occupancyPct;
+  }
+}
+
+// flip this to true to write changes to the database;
+// flip it to false for a dry run that just logs to console
+const write = false;
+
+const setFacilityCapacity = async ({
+  facilityDoc,
+  scenarioId,
+  modelVersionDoc,
+}) => {
+  let { modelInputs } = facilityDoc.data();
+  let modelVersionId;
+
+  if (modelVersionDoc) {
+    modelInputs = modelVersionDoc.data();
+    modelVersionId = modelVersionDoc.id;
+  }
+
+  const capacity = calculateFacilityCapacity(modelInputs);
+  if (capacity) {
+    addFacilityCapacity({
+      facilityId: facilityDoc.id,
+      facilityCapacity: capacity,
+      modelInputs,
+      scenarioId,
+      write,
+      modelVersionId,
+    });
+  } else {
+    console.log(
+      `unable to calculate capacity for facility ${facilityDoc.id}${
+        modelVersionId ? ` version ${modelVersionId}` : ""
+      }`,
+      modelInputs,
+    );
+  }
+};
+
+// eslint-disable-next-line react/display-name
+export default () => {
+  const migrate = async () => {
+    const scenarios = await getMigrationScenarios();
+    scenarios.forEach(async (scenarioDoc) => {
+      const scenarioId = scenarioDoc.id;
+      const facilities = await getMigrationFacilities(scenarioId);
+      facilities.forEach(async (facilityDoc) => {
+        setFacilityCapacity({ facilityDoc, scenarioId: scenarioId });
```

Comment:
Why do you have the explicit key here but not below on line 73?

Reply:
this call updates `facility.modelInputs` (the "current" input version that we store in the facility document itself) ... the one on line 73 updates each of the documents in this facility's `modelVersions` collection.